packages/api/src/feature_flag/controller.test.ts

```diff
@@ -0,0 +1,112 @@
+import request from "supertest";
+import type { NextFunction, Request } from "express";
+import { logger } from "@stela/logger";
+import { db } from "../database";
+import { app } from "../app";
+import { extractUserIsAdminFromAuthToken } from "../middleware";
+import type { FeatureFlagRequest } from "./models";
+
+jest.mock("../database");
+jest.mock("../middleware");
+jest.mock("@stela/logger");
+
+describe("GET /feature-flags", () => {
+  const agent = request(app);
+
+  const loadFixtures = async (): Promise<void> => {
+    await db.sql("fixtures.create_test_feature_flags");
+  };
+
+  const clearDatabase = async (): Promise<void> => {
+    await db.query("TRUNCATE feature_flag CASCADE");
+  };
+
+  beforeEach(async () => {
+    await clearDatabase();
+    await loadFixtures();
+    (extractUserIsAdminFromAuthToken as jest.Mock).mockImplementation(
+      (req: Request, __, next: NextFunction) => {
+        (req.body as FeatureFlagRequest).admin = false;
+        next();
+      }
+    );
+  });
+
+  afterEach(async () => {
+    jest.restoreAllMocks();
+    await clearDatabase();
+  });
+
+  test("should log the error if the database call fails", async () => {
+    const testError = new Error("SQL error");
+    jest.spyOn(db, "sql").mockRejectedValueOnce(testError);
+
+    await agent.get("/api/v2/feature-flags").expect(500);
+    expect(logger.error).toHaveBeenCalled();
+  });
+
+  test("should log the error if the database call fails when calling user is admin", async () => {
+    (extractUserIsAdminFromAuthToken as jest.Mock).mockImplementation(
+      (req: Request, __, next: NextFunction) => {
+        (req.body as FeatureFlagRequest).admin = true;
+        next();
+      }
+    );
+    const testError = new Error("SQL error");
+    jest.spyOn(db, "sql").mockRejectedValueOnce(testError);
+
+    await agent.get("/api/v2/feature-flags").expect(500);
+    expect(logger.error).toHaveBeenCalled();
+  });
+
+  test("should log the error if the request is invalid", async () => {
+    (extractUserIsAdminFromAuthToken as jest.Mock).mockImplementation(
+      (_req: Request, __, next: NextFunction) => {
+        next();
+      }
+    );
+
+    await agent.get("/api/v2/feature-flags").expect(500);
```

Comment:
For an invalid request we should expect a 400 error

Reply:
This scenario would not be possible because extractUserIsAdminFromAuthToken would always return a boolean request.body.admin value. Maybe it would be best to just remove this test?

Reply:
Done, thanks :)